st2api/tests/unit/controllers/v1/test_executions_filters.py

```diff
@@ -194,6 +194,22 @@ def test_filters(self):
             self.assertGreater(len(response.json), 0)
             self.assertGreater(int(response.headers['X-Total-Count']), 0)
 
+    def test_advanced_filters(self):
+        excludes = ['parent', 'timestamp', 'action', 'liveaction', 'timestamp_gt',
+                    'timestamp_lt', 'status']
+        for param, field in six.iteritems(ActionExecutionsController.supported_filters):
+            if param in excludes:
+                continue
+
+            value = self.fake_types[0]
+            for item in field.split('.'):
+                value = value[item]
+            response = self.app.get('/v1/executions?filter=%s:%s' % (field, value))
+            self.assertEqual(response.status_int, 200)
+            self.assertIsInstance(response.json, list)
+            self.assertGreater(len(response.json), 0)
+            self.assertGreater(int(response.headers['X-Total-Count']), 0)
```

Comment:
1) Can you add another unit test that show the exception thrown if there are more than max split in the filter value?  For example `/v1/executions?filter=k1:v1,k2:v2`.
2) Also, is multiple filters supported?  For example `/v1/executions?filter=k1:v1&filter=k2:v2`

Reply:
Multiple filters are not supported. There's no way to express anything but OR logic this way and we've already been bitten by that in criteria so I don't want to make the same mistake again. Filter will always going to be a string user composes (semi-)manually so it makes sense to send it to API as a single string.

I'm going to make sure we output a readable exception for 1.

Reply:
Actually, I could not find a proper way to handle `/v1/executions?filter=k1:v1,k2:v2` case at the moment. My gut feeling tells me that we should treat `,` same as space (a delimiter), but at that point I simply don't know.

I've made advanced filter field less noticeable in UI and I'm think we should release it as is to get some early feedback and see where to go next.

Reply:
If a user calls something like this /v1/executions?filter=k1:v1,k2:v2, then do we throw errors? unit test?